packages/validation/src/validators/ValidationFunctions.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * A list of functions that are used to validate primitive fields for LionWeb conformance.
+ * Used in the LionWebSyntaxValidator.
+ */
+import { JsonContext } from "../json/JsonContext.js"
+import { Language_PropertyValue_Issue } from "../issues/LanguageIssues.js"
+import {
+    Syntax_IdFormat_Issue,
+    Syntax_KeyFormat_Issue, Syntax_PropertyNullIssue,
+    Syntax_SerializationFormatVersion_Issue,
+    Syntax_VersionFormat_Issue
+} from "../issues/SyntaxIssues.js"
+import { ValidationResult } from "./generic/ValidationResult.js"
+import { PropertyDefinition } from "./generic/ValidationTypes.js"
+
+/**
+ * Check whether `id` is a valid LionWeb id.
+ * @param value     The `value` to be checked.
+ * @param result    Any validation issues found will be put into this object.
+ * @param context   The context for the error message in errors.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function validateId<String>(value: String, result: ValidationResult, context: JsonContext): void  {
+    const idString: string = "" + value
+    const egexp = /^[a-zA-Z0-9$_-][a-zA-Z0-9$_-]*$/
+    if (!egexp.test(idString)) {
+        result.issue(new Syntax_IdFormat_Issue(context, idString))
+    }
+}
+
+/**
+ * Check whether `key` is a valid LionWeb key.
+ * @param value     The `key` to be checked.
+ * @param result    Any validation issues found will be put into this object.
+ * @param context   The context for the error message in errors.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function validateKey<String>(value: String, result: ValidationResult, context: JsonContext): void {
+    const keyString: string = "" + value
+    const egexp = /^[a-zA-Z0-9$_-][a-zA-Z$_0-9-]*$/
```

Comment:
Same as above. But more importantly: why aren't these regexpes not **exactly** the same? (They're [already] equivalent.)